### PR TITLE
Various size optimisations

### PR DESCRIPTION
**Pull request by RReverser**
Improved bundle sizes using a range of techniques (each optimisation is described in a separate commit).

There are few more potential optimisations, but they provide less drastic results.

Gzipped sizes:
ESM: 564 -> 511 (-9%)
CJS: 617 -> 530 (-15%)
UMD: 676 -> 617 (-9%)

### CODE REVIEW

**Comment by RReverser** [comment]
@surma ping?

File: rollup-plugins/export-in-place.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+export default function() {
+  return {
+    name: "export-in-place",
+
+    renderChunk(code) {
+      const ast = this.parse(code);
+      // Assert for the AST shape.
+      // 1) We should have only two declarations at top level.
+      assert.strictEqual(ast.body.length, 2);
```
Comment (surma) [inline rollup-plugins/export-in-place.js]: This means the plugin will throw when the AST has an unexpected shape, no?

I’d prefer to log a warning with `this.warn()` when that happens, as it’s still valid code.

**Review by surma — CHANGES_REQUESTED** [review]
Sorry, forgot about this.

Kinda hilarious that you wrote a dedicated plugin, but it works! Just one thing I’d like to change.

**Comment by RReverser** [comment]
> Kinda hilarious that you wrote a dedicated plugin, but it works!

Yeah I'd much prefer this to be the default, but I'm not sure yet which project it belongs to yet and, well, need to describe the usecases for the issue report. Code golfing might not be reasonable enough :)

File: rollup-plugins/export-in-place.js (hunk 1 above)
Comment (RReverser) [reply rollup-plugins/export-in-place.js]: Sure; just wanted to make sure that it's easy to catch a case if output format ever changes.

Comment (RReverser) [reply rollup-plugins/export-in-place.js]: Added meaningful assert messages and a try-catch to just warn instead and fall back to the original source.

**Review by surma — APPROVED** [review]

**Comment by RReverser** [comment]
Submitted upstream issue in Terser to see if they would be interested: https://github.com/terser/terser/issues/641

**Pull request merged**